File: toastz/src/main/java/com/duyangs/toastz/ToastZLib.java
package com.duyangs.toastz;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.annotation.NonNull;

/**
 * <p>Project:ToastZ</p>
 * <p>Package:com.duyangs.toastz</p>
 * <p>Description:</p>
 * <p>Company:</p>
 *
 * @author duyangs
 * @date 2017/04/19 0019
 */
public class ToastZLib {

    @SuppressLint("StaticFieldLeak")
    private static Context context;

    private ToastZLib() {
        throw new UnsupportedOperationException("u can't instantiate me...");
    }

    /**
     * 初始化工具类
     *
     * @param context 上下文
     */
    public static void init(@NonNull Context context) {
        ToastZLib.context = context.getApplicationContext();
    }

    /**
     * 获取ApplicationContext
     *
     * @return ApplicationContext
     */
    public static Context getContext() {
        if (context != null) return context;
        throw new NullPointerException("u should init first");
    }
}

File: toastz/src/main/java/com/duyangs/toastz/ToastZLib.kt
package com.duyangs.toastz

import android.annotation.SuppressLint
import android.content.Context
import androidx.annotation.NonNull

@SuppressLint("StaticFieldLeak")
/**
 * <p>Project:ToastZ</p>
 * <p>Package:com.duyangs.toastz</p>
 * <p>Description:Initialization class</p>
 * <p>Company:</p>
 *
 * @author duyangs
 * @date 2018/06/04
 */
object ToastZLib {

    private var mContext: Context? = null

    /**
     * 初始化工具类
     *
     * @param context 上下文
     */
    fun init(context : Context ) {
        mContext = context.applicationContext
    }

    /**
     * 获取ApplicationContext
     *
     * @return ApplicationContext
     */
    fun getContext(): Context {
        if (mContext != null)
            return mContext as Context
        else
            throw NullPointerException("u should init first")
    }
}
